solvebio/query.py

```diff
@@ -292,10 +292,29 @@ def __repr__(self):
         if len(self) == 0:
             return 'Query returned 0 results.'
 
-        return '\n%s\n\n... %s more results.' % (
-            tabulate(list(self._buffer[0].items()), ['Fields', 'Data'],
-                     aligns=['right', 'left'], sort=True),
-            pretty_int(len(self) - 1))
+        placeholder = 'many more results (total unknown)'
+
+        if getattr(self, '_is_join', False):
+            return '\n%s\n\n... %s.' % (
+                tabulate(list(self._buffer[0].items()), ['Fields', 'Data'],
+                         aligns=['right', 'left'], sort=True),
+                placeholder)
+        elif not getattr(self, '_output_format', False):
```

Comment:
I think this may be wrong now since _output_format defaults to json

Reply:
All is good, however, we can use `hasattr` instead of `getattr`  because we do not need value.

Reply:
Ok, I did notice that file-based queries on JSON files now print the rows and not the JSON objects

```
f = Object.retrieve("XYZ")
f.query()

|          Fields | Data                                                                                            |
|-----------------+-------------------------------------------------------------------------------------------------|
|   configuration | 1                                                                                            |
|           debug | False                                                                                           |
| output_dir_path | xyz                                                                               |
|          sample | {...} |                                                             |
|         version | 3
```

Now gives:

```
|   Row | Data                                                                                                      |
|-------+-----------------------------------------------------------------------------------------------------------|
|     0 | {u'sample': {... |

```

Reply:
Ohh, yes! I changed that to be in accordance with the new file preview in the UI. Please look at this[PR](https://github.com/solvebio/www.solvebio.com/pull/1953). 

It can be changed easily if you want.

Reply:
![image](https://user-images.githubusercontent.com/13609168/109855531-af015b00-7c58-11eb-83e4-05bc126170bd.png)

Reply:
Ok, I see that. I feel like it should be similar to how dataset.query() works, at least for JSON files. Doesn't really matter that much though. I'm ok with how it is now.